lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -417,11 +417,19 @@ public function processParameterValue($value)
             return $value;
         }
 
-        if (is_object($value) && $this->_em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
-            $value = $this->_em->getUnitOfWork()->getSingleIdentifierValue($value);
+        if (is_object($value)) {
+            try {
+                $this->_em->getMetadataFactory()->getMetadataFor(ClassUtils::getClass($value));
+            } catch (\Exception $e) {
```

Comment:
Then I also need to catch `Doctrine\ORM\Mapping\MappingException` exceptions because when I run the tests, all the exceptions caught are from this class.